compat/src/suspense.js

```diff
@@ -41,79 +42,14 @@ options.unmount = function(internal) {
 	if (oldUnmount) oldUnmount(internal);
 };
 
-/**
- *
- * @param {import('./internal').Internal} internal
- * @param {import('./internal').PreactElement} detachedParent
- * @param {import('./internal').PreactElement} parentDom
- */
-function detachedClone(internal, detachedParent, parentDom) {
-	if (internal) {
-		if (internal._component && internal._component.__hooks) {
-			internal._component.__hooks._list.forEach(effect => {
-				if (typeof effect._cleanup == 'function') effect._cleanup();
-			});
-
-			internal._component.__hooks = null;
-		}
-
-		internal = assign({}, internal);
-		if (internal._component != null) {
-			if (internal._component._parentDom === parentDom) {
-				internal._component._parentDom = detachedParent;
-			}
-			internal._component = null;
-		}
-
-		if (internal._children) {
-			internal._children = internal._children.map(child =>
-				detachedClone(child, detachedParent, parentDom)
-			);
-		}
-	}
-
-	return internal;
-}
-
-/**
- * @param {import('./internal').Internal} internal
- * @param {import('./internal').PreactElement} detachedParent
- * @param {import('./internal').PreactElement} originalParent
- */
-function removeOriginal(internal, detachedParent, originalParent) {
```

Comment:
We no longer reset the `_vnodeId` property or add the `FORCE_UPDATE` flag to all suspended children so any vnode equality or `sCU` lifecycles under the Suspense node are still respected. We do still call `.forceUpdate` on the component that suspended (i.e. threw the promise) to ensure it gets rendered, but any intermediate children between it and Suspense won't necessarily be re-rendered in this new implementation.

I think we gone back and forth on this. If anyone knows of any use cases where we must re-render intermediate children, let me know!

Reply:
FYI, this is PR #2661 added the `removeOriginal` function, for additional context